### PR TITLE
Store generated p2p private key in a database dir and reuse it if none user key is supplied

### LINKED ISSUE #2209: Persist generated NodeID PrivKey when not supplied
### Description

When no `--p2p-private-key-file` option is supplied Teku generates a random KeyPair for NodeID every launch 
It would be more convenient if the private key generated on the first run is persisted and used on subsequent runs when `--p2p-private-key-file` is not set explicitly.

### PR DESCRIPTION
**Pull request by Nashatyrev**
## PR Description

Store generated p2p private key in a database dir and reuse it if none user key is supplied

TODO: 
- [x] add unit tests

## Fixed Issue(s)

Fixes #2209 

## Documentation

- [x] I thought about documentation and added the `documentation` label to this PR if updates are required.

### CODE REVIEW

**Comment by Nashatyrev** [comment]
Questions: 
- is it fine to store private key (even though generated) this plain way? 
- is the file location fine? 
- should we probably log message to console that generated key was created/loaded?

**Comment by ajsutton** [comment]
I don't think there's any real option other than to store the key in plain text since we need to be able to read it at startup anyway - if we encrypted it the password would have to be available in plain text anyway.

File location makes sense to me.

Yes, I think it's worth logging a message to console to say the key was generated.

We'll need to make sure we store the ENR sequence number at the same time as we enable this I suspect otherwise things will get out of sync quickly - I think Prysm in particular is pretty strict about rejecting ENRs when the sequence number goes backwards which may cause us connectivity problems.  Could commit this with a feature toggle to disable it until we get the sequence number also stored.

**Comment by Nashatyrev** [comment]
>We'll need to make sure we store the ENR sequence number at the same time as we enable this I suspect otherwise things will get out of sync quickly - I think Prysm in particular is pretty strict about rejecting ENRs when the sequence number goes backwards which may cause us connectivity problems. Could commit this with a feature toggle to disable it until we get the sequence number also stored.

I will merge this PR together with the fix for #1961

File: logging/src/main/java/tech/pegasys/teku/logging/StatusLogger.java [hunk 1]
```diff
@@ -155,4 +155,14 @@ public void eth1ServiceDown(final long interval) {
   public void eth1AtHead() {
     log.info("Eth1 tracker successfully caught up to chain head");
   }
+
+  public void usingGeneratedP2pPrivateKey(final String keyFile, final boolean justGenerated) {
+    if (justGenerated) {
+      log.info(
+          "Generated new p2p private key. It was stored and will be reused on next run if no private key option is supplied: "
+              + keyFile);
```
Comment (ajsutton) [inline logging/src/main/java/tech/pegasys/teku/logging/StatusLogger.java]:
```suggestion
          "Generated new p2p private key and storing in: "
              + keyFile);
```

**Review by ajsutton — APPROVED** [review]
LGTM.

File: logging/src/main/java/tech/pegasys/teku/logging/StatusLogger.java (hunk 1 above)
Comment (Nashatyrev) [reply logging/src/main/java/tech/pegasys/teku/logging/StatusLogger.java]: Thanks 👍

**Comment by Nashatyrev** [comment]
Merged this PR into #2715 and made use of `KeyValueStore` for storing generated P2p key. 
Closing this. Please review #2715

**Comment by Nashatyrev** [comment]
@MadelineMurray please update documentation related record to refer #2715

**Pull request closed without merging**